### ISSUE TITLE
gulp buildcontrol:heroku : Failing

### ISSUE THREAD

**Issue by jofftiquez** [angular-fullstack/generator-angular-fullstack#2009]
I was trying to deploy my sample angular-fullstack app to heroku so I did this `gulp buildcontrol:heroku`. But Im getting this error. 

> [14:17:19] Requiring external module babel-register
> [14:17:21] Using gulpfile /var/www/html/WiM/gulpfile.babel.js
> [14:17:21] Starting 'buildcontrol:heroku'...
> Running "buildcontrol:heroku" (buildcontrol) task
> 
> fatal: 'heroku' does not appear to be a git repository
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> Warning: fatal: 'heroku' does not appear to be a git repository
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> Use --force to continue.
> 
> Aborted due to warnings.

Im pretty sure I added the git url from heroku as a remote on my local git repo.

Item | Version
----- | -----
generator-angular-fullstack | 3.7.5
Node | 5.10.1
npm | 3.8.7
Operating System | Linux Mint

Item | Answer
----- | -----
Transpiler | Babel
Markup | HTML
CSS | CSS 
Router | ui-router
Build Tool |  Gulp
Client Tests |  Mocha
DB | MongoDB 
Auth | Y

**Comment by gpessa** [comment]
same problem here

**Comment by brentlemons** [comment]
`buildcontrol:heroku` seems to be broken for gulp, but I think the actual problem is `yo angular-fullstack:heroku` does not create the heroku project or git remote.

It's a bit of a pain in the ass, but I got it to work.

Starting from your project folder:
`heroku create <your-project>`
`heroku addons:create mongolab`
`gulp build`
`cd dist`
`git init`
`heroku git:remote -a <your-project>`
`gulp buildcontrol:heroku`

If I get a chance, I'll look at creating a PR to fix the issue.

**Comment by jofftiquez** [comment]
@brentlemons - Exact thoughts. What I did to make it work is a bit weird though I think. I created the herokuapp from heroku website. Then I added the remote on the root of the project, not in the `dist` directory. After that I added the `git url` directly in the Gulfile.js under `buildcontrol.heroku.options.remote` I replaced the value 'heroku' with the `git url`. I dont really know if it's the right way since this method is not mentioned in the documentation AFAIR, but it worked. 

TL;DR

1. Create herokuapp from heroku website.
2. `git init` in root of project
3. `git remote add heroku <the git url from heroku website>`
4. Added `<the git url from heroku website>` to `Gulpfile.js` under `buildcontrol.heroku.options.remote` object.
5. `gulp buildcontrol:heroku`
6. It worked. IKR?

**Comment by nstuyvesant** [comment]
Would love to see the grunt and grunt-build-control devDependencies go away in the future so that it's pure gulp.
